corehq/apps/cloudcare/static/cloudcare/js/formplayer/router.js

```diff
@@ -275,6 +275,12 @@ hqDefine("cloudcare/js/formplayer/router", [
     });
 
     FormplayerFrontend.on("breadcrumbSelect", function (index) {
+        if (FormplayerFrontend.formInProgress) {
+            const userConfirmed = window.confirm("You have a form in progress. Are you sure you want to navigate away?");
+            if (!userConfirmed) {
+                return;
+            }
+        }
         FormplayerFrontend.trigger("clearForm");
```

Comment:
Would it work to hook into `clearForm` instead of `breadcrumbSelect`?  I don't know quite what the new persistent menu will use, but it seems fairly safe to assume `clearForm` will be in both versions.